docs/data/data-grid/editing/ValueParserSetterGrid.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  GridColDef,
+  GridValueGetterParams,
+  GridValueSetterParams,
+  GridCellValue,
+} from '@mui/x-data-grid';
+
+function getFullName(params: GridValueGetterParams) {
+  return `${params.row.firstName || ''} ${params.row.lastName || ''}`;
+}
+
+function setFullName(params: GridValueSetterParams) {
+  const [firstName, lastName] = params.value!.toString().split(' ');
+  return { ...params.row, firstName, lastName };
+}
+
+function parseFullName(value: GridCellValue) {
+  return String(value).toUpperCase();
+}
```

Comment:
The demo seems a bit weird with the value parser putting everything in uppercase whereas the initial values are not upper case. Maybe a parser that only uppercases the first letter will seems less buggy

![valueParser](https://user-images.githubusercontent.com/45398769/157228573-bae119af-77d2-48b8-97db-7573fd2574f7.gif)
![valueParser](https://user-images.githubusercontent.com/45398769/157228607-0271bcdb-e1b7-400f-bc8d-e0d2e43b2468.gif)